src/components/headerSearch.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from "react";
+import Mark from "mark.js";
+import { IonButton, IonButtons, IonIcon, IonInput, IonItemDivider } from "@ionic/react";
+import { arrowDown, arrowUp } from 'ionicons/icons';
+
+const HeaderSearch = () => {
+  const [needle, setNeedle] = useState<string>('');
+  const [marks, setMarks] = useState<NodeListOf<HTMLElement>>(document.querySelectorAll('mark'));
+  const [currentIndex, setCurrentIndex] =  useState<number>(0);
+  const doc = document.querySelector('.akoma-ntoso') as HTMLElement;
+  let mark: any;
+
+  const style = {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    margin: '10px auto',
+    paddingLeft: '0.5rem'
+  }
+
+  useEffect(() => {
+    search()
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [needle]); 
+
+  const jumpTo = () => {
+    if (marks.length > 0) {
+      const currentMark = marks[currentIndex];
+      marks.forEach(mark => mark.classList.remove('current'));
+      if (currentMark) {
+        currentMark.classList.add('current');
+        currentMark.scrollIntoView({
+          behavior: "smooth",
+          inline: "start"
+        });
+      }
+    }
+  }
+    
+  const jumpToPrevious = () => {
+    if (marks.length > 0) {
+      setCurrentIndex(currentIndex - 1);
```

Comment:
It's safe to use `currentIndex--`, so that we don't change setCurrentIndex twice.

Reply:
This will also fix the jumping bug, because after calling 'setCurrentIndex`, the value of `currentIndex` is not immediately updated.

So, fiddle `currentIndex` until it's what you need, then call `setCurrentIndex(currentIndex)` once.